macros/string_index_based_replacer.go

```diff
@@ -102,3 +102,21 @@ func (s *stringIndexBasedReplacer) getTemplate(url string) urlMetaTemplate {
 	}
 	return template
 }
+
+func (s *stringIndexBasedReplacer) ReplaceBytes(result *strings.Builder, url string, macroProvider *MacroProvider) {
+	tmplt := s.getTemplate(url)
+	currentIndex := 0
+	delimLen := len(delimiter)
+	for i, index := range tmplt.startingIndices {
+		macro := url[index : tmplt.endingIndices[i]+1]
+		// copy prev part
+		result.WriteString(url[currentIndex : index-delimLen])
+		value := macroProvider.GetMacro(macro)
+		if value != "" {
+			result.WriteString(value)
+		}
+		currentIndex = index + len(macro) + delimLen
+	}
+	result.WriteString(url[currentIndex:])
```

Comment:
Can you add test coverage for this function?

Reply:
Done